include/boost/geometry/util/type_traits.hpp

```diff
@@ -0,0 +1,323 @@
+{};
+template<typename Trait>
+struct conjunction<Trait>
+    : Trait
+{};
+template <typename Trait, typename ...Traits>
+struct conjunction<Trait, Traits...>
+    : std::conditional_t<Trait::value, conjunction<Traits...>, Trait>
+{};
+
+
+// C++17
+template <typename ...>
+struct disjunction
+    : std::false_type
+{};
+template <typename Trait>
+struct disjunction<Trait>
+    : Trait
+{};
+template <typename Trait, typename ...Traits>
+struct disjunction<Trait, Traits...>
+    : std::conditional_t<Trait::value, Trait, disjunction<Traits...>>
+{};
+
+
+// C++17
+template <typename Trait>
+struct negation
+    : bool_constant<!Trait::value>
+{};
+
+
+/*
+template <typename ...Traits>
+using and_ = conjunction<Traits...>;
+
+template <typename ...Traits>
+using or_ = disjunction<Traits...>;
+
+template <typename Trait>
+using not_ = negation<Trait>;
+*/
+
+
+template <typename T>
+struct is_geometry
+    : bool_constant<! std::is_void<typename tag<T>::type>::value>
+{};
+
+template <typename T>
+struct is_not_geometry
+    : std::is_void<typename tag<T>::type>
+{};
+
+
+template <typename T>
+struct is_point
+    : std::is_same<point_tag, typename tag<T>::type>
+{};
+
+template <typename T>
+struct is_multi_point
+    : std::is_same<multi_point_tag, typename tag<T>::type>
+{};
+
+template <typename T>
+struct is_pointlike
+    : std::is_base_of<pointlike_tag, typename tag<T>::type>
+{};
+
+
+template <typename T>
+struct is_segment
+    : std::is_same<segment_tag, typename tag<T>::type>
+{};
+
+template <typename T>
+struct is_linestring
+    : std::is_same<linestring_tag, typename tag<T>::type>
+{};
+
+template <typename T>
+struct is_multi_linestring
+    : std::is_same<multi_linestring_tag, typename tag<T>::type>
+{};
+
+template <typename T>
+struct is_polylinear
+    : std::is_base_of<polylinear_tag, typename tag<T>::type>
+{};
+
+template <typename T>
+struct is_linear
+    : std::is_base_of<linear_tag, typename tag<T>::type>
+{};
+
+
+template <typename T>
+struct is_box
+    : std::is_same<box_tag, typename tag<T>::type>
+{};
+
+template <typename T>
+struct is_ring
+    : std::is_same<ring_tag, typename tag<T>::type>
+{};
+
+template <typename T>
+struct is_polygon
+    : std::is_same<polygon_tag, typename tag<T>::type>
+{};
+
+template <typename T>
+struct is_multi_polygon
+    : std::is_same<multi_polygon_tag, typename tag<T>::type>
+{};
+
+template <typename T>
+struct is_polygonal
+    : std::is_base_of<polygonal_tag, typename tag<T>::type>
+{};
+
+template <typename T>
+struct is_areal
+    : std::is_base_of<areal_tag, typename tag<T>::type>
+{};
+
+
+template <typename T>
+struct is_polysegmental
+    : bool_constant<is_polylinear<T>::value || is_polygonal<T>::value>
+{};
+
+
+template <typename T>
+struct is_multi
+    : std::is_base_of<multi_tag, typename tag<T>::type>
+{};
+
+
+template <typename T>
+struct is_multi_element
+    : bool_constant<is_point<T>::value || is_linestring<T>::value || is_polygon<T>::value>
+{};
```

Comment:
Here is another group, single geometries which can be elements of a multi geometry. I didn't have an idea about a name so the one I used is more straightforward than the other ones I mentioned. I'm open to suggestion.

Reply:
Fine for me, it's clear.

Reply:
The names are fine by me too. I do not know if they make a lot of geometric sense but they are consistent and clear. So I have no objection.